lfs/client.go

```diff
@@ -125,8 +126,15 @@ func Download(oid string) (io.ReadCloser, int64, error) {
 		return nil, 0, err
 	}
 	LogTransfer("lfs.data.download", res)
+	
+	buf, _ := ioutil.ReadAll(res.Body)
+	body := myReader{bytes.NewBuffer(buf)}	
```

Comment:
The connection isn't guaranteed to be kept alive unless the *entire* body is read and then closed. See https://github.com/golang/go/issues/5645

I think for batch to work we need to handle multiple connections. 

Maybe we could check the response size and not copy it (and force the re-auth) if it's under some limit